main.tf

```
@@ -36,3 +42,30 @@ resource "aws_security_group" "elasticache" {
     protocol    = "tcp"
   }
 }
+
+resource "aws_elasticache_replication_group" "default" {
+  count = var.replication_group ? 1 : 0
+
+  at_rest_encryption_enabled    = var.at_rest_encryption
+  auth_token                    = var.in_transit_encryption ? var.auth_token : null
+  auto_minor_version_upgrade    = true
+  automatic_failover_enabled    = var.num_cache_nodes > 1 ? true : false
+  availability_zones            = var.availability_zones
+  engine                        = var.engine
+  engine_version                = var.engine_version
+  kms_key_id                    = var.at_rest_encryption ? var.kms_key_arn : null
```

This can just be like this if the default is `null`:
```suggestion
  kms_key_id                    = var.kms_key_arn
```